google/cloud/storage/client.py

```diff
@@ -1480,6 +1530,57 @@ def list_buckets(
             retry=retry,
         )
 
+    def restore_bucket(
+        self,
```

Comment:
Do we want to add the optional query param `projection`? Seems like the API accepts 
`projection=noAcl` or `full`. It would be nice if there were ready to publish docs to refer to.

Reply:
Added